### DISCUSSION TITLE
How to add tests

### DISCUSSION THREAD

**Discussion by yaten2302** [anuraghazra/github-readme-stats#3466] [Q&A] [answered]
Hi @rickstaa, @qwerty541, 

I was having a doubt that how should the test be added? Like for this PR - [card width of gist card](https://github.com/anuraghazra/github-readme-stats/pull/3405), how should I add the tests for this?

**Comment by rickstaa** [comment]
@yaten2302, thanks for contributing to GRS 🎉. The tests of the gist card are found in the [tests/renderGistCard.test.js](https://github.com/anuraghazra/github-readme-stats/blob/master/tests/renderGistCard.test.js) file. We use [jest](https://jestjs.io/) as our testing provider. You can check out the [CONTRIBUTING.md](https://github.com/anuraghazra/github-readme-stats/blob/master/CONTRIBUTING.md#local-development) to see how to run and debug tests.

For your specific use case since the `card_width` parameter is already present in the stats card, you can use the tests defined there:

https://github.com/anuraghazra/github-readme-stats/blob/b0bb994ad36d7c187e2f934e36e34978e9de5e21/tests/renderStatsCard.test.js#L78-L122

**Reply by yaten2302** [reply]
@rickstaa, can I simply copy these tests to the `tests/renderGistCard.test.js` file, after changing the `renderStatsCard` to `renderGistCard`?

**Reply by rickstaa** [reply, marked as answer]
Thats correct. You might, however, need to change the width values to make them pass.
